iTunes Search/iTunes Search/Model Controller/SearchResultController.swift

```diff
@@ -0,0 +1,57 @@
+//
+//  SearchResultController.swift
+//  iTunes Search
+//
+//  Created by Mitchell Budge on 5/14/19.
+//  Copyright © 2019 Mitchell Budge. All rights reserved.
+//
+
+import Foundation
+
+class SearchResultController {
+    let baseURL = URL(string: "https://itunes.apple.com/search")!
+    var results: [SearchResult] = []
+    enum HTTPMethod: String {
+        case get = "GET"
+        case put = "PUT"
+        case post = "POST"
+        case delete = "DELETE"
+    }
+    
+    func performSearch(with searchTerm: String, and resultType: ResultType, completion: @escaping (Error?) -> Void) {
+        //let searchResultsURL = baseURL.appendingPathComponent("results")
+        var urlComponents = URLComponents(url: baseURL, resolvingAgainstBaseURL: true)
+        let searchQueryItem = URLQueryItem(name: "term", value: searchTerm)
+        let resultTypeQueryItem = URLQueryItem(name: "entity", value: resultType.rawValue)
+        urlComponents?.queryItems = [searchQueryItem, resultTypeQueryItem]
+        guard let formattedURL = urlComponents?.url else {
+            completion(nil)
```

Comment:
We don't want to pass nil here cause wherever we call this will then assume "do nothing". Instead we want to create an error, one way is to create an NSError()

`completion(NSError())`